### DISCUSSION TITLE
Provieder 6591 entbranden und freetzten, wie gehe ich vor?

### DISCUSSION THREAD

**Discussion by Conan179** [Freetz-NG/freetz-ng#767] [Allgemein]
Hi, ich möchte meine Provider 6591 endbranden und mit freetz versehen, wie gehe ich am besten vor? Was ich gerne machen würde, wäre ein backup zu machen, wo ich wieder einspielen kann, wen was schief läuft.
Sie soll keine internet box sein, sondern als mesh router arbeiten, wen möglich die daten der Kanäle überwachen.

**Comment by fda77** [comment]
Es gibt leider keine Anleitung in der alles zusammengefasst ist. Du musst dir das in den verschiedenen Foren zusammensuch

**Comment by AkteX** [comment]
Dieses Thema wird im Digital Eliteboard behandelt. Eine Rückkehr zu einer Provider FW macht keinen Sinn, da in diesem Fall der Inhalt des Bootloaders zurückgepatcht werden müsste. Mit gepachten Bootloader kannst Du entweder eine Original AVM oder Freetz-FW verwenden.

**Reply by Conan179** [reply]
> Eine Rückkehr zu einer Provider FW macht keinen Sinn,

Es geht mir nur darum, ein backup zuhaben, wen ich mit freetz was faösch gemacht habe und die box nicht mehr startet ich es drauf hauen kann und sie startet wieder, da es ja für die Kabel Boxen kein Recovery.exe gibt.

**Reply by AkteX** [reply]
Dies würde Dir nur etwas nützen, wenn der Bootloader nicht auf AVM umgestellt wird. Darüber hinaus kannst Du die Backup Möglichkeit des DEB Verfahrens nicht verwenden, um die Original FW zu sichern. Es werden nur die aktiven Flash Blöcke gesichert und dort befindet sich die zuvor geflashte FW.
Wenn Du den Bootloader auf AVM gepatcht hast, kannst Du eine Original AVM FW oder Freetz-FW flashen. Das geht immer über EVA-FTP bei einem Bootloader, welcher nicht älter ist, als Sept. 2020. Das ist quasi eine Recovery Möglichkeit.

**Reply by fda77** [reply]
Was man auch machen könnte: Die Provider FW auftreiben und einfach damit freetz erstellen.
Die müsste jemand mit provider-fw auftreiben, zb durch mitschneiden der juis-anfrage um die parameter zu erhlaten - falls es nicht nur via tr069 oder noch schlimmer tr369

**Reply by AkteX** [reply]
Im Feuerfritz Readme gibt es das Kapitel: Temporarily modify firmware_version. Damit könnte man eine FW erzeugen, welche auf gebrandeten Boxen lauffähig ist.

**Reply by fda77** [reply]
> Feuerfritz Readme gibt es das Kapitel: Temporarily modify firmware_version

Wo soll das sein, Ich hab es nicht gefunden

**Reply by AkteX** [reply]
https://bitbucket.org/fesc2000/ffritz/src/6591/README-6591.md

Temporarily modify firmware_version
To generate a firmware image which will boot on a branded box:

Copy the file user-oem.patch to the atom/arm directories and generate a modified update image:

cp user-oem.patch puma7/atom/
cp user-oem.patch puma7/arm/
make
This image can be installed on a branded box, but several features might not work (DVB-C, SIP, ..?). In any case, an automatic firmware update will not be offered .. which is good, as this would install an image that can't boot (as it obviously does not contain my patch).

For each new firmware version, you need to re-generate a patched image and install it manually.

You can try to activate missing features by doing the next step, but then

! don't ever install an update offered by AVM via the GUI, and never activate automatic update !

**Reply by fda77** [reply]
Der "ffritz" ist der "feuerfritz" :) Muss man wissen...
Ein 3. Name scheint "fesc2000" zu sein, ich glaub "fesc" hab ich auch mal gelesen 

Der user-oem.patch ist jedenfalls nichts grosses: https://bitbucket.org/fesc2000/ffritz/src/6591/user-oem.patch

```
--- etc/init.d/rc.conf
+++ etc/init.d/rc.conf
@@ -392,15 +392,17 @@
 ##########################################################################################
 ## OEM Ermitteln
 ##########################################################################################
-OEM_tmp=`cat $CONFIG_ENVIRONMENT_PATH/firmware_version`
-if [ -z "${OEM_tmp}" ] ; then
-OEM_tmp=$CONFIG_OEM_DEFAULT
-fi
-OEM=${OEM_tmp%,*}
-OEM_DEFAULT_INDEX=${OEM_tmp#*,}
-if [ "$OEM_DEFAULT_INDEX" = "$OEM" ] ; then
+OEM=$CONFIG_OEM_DEFAULT
 OEM_DEFAULT_INDEX=""
-fi
```

Dh es wird nur in 1 Script das branding/oem geändert - JE CPU
Aber bei den Puma Geräten wird von Freetz nur der Atom Teil bearbeitet, Arm ist unverändert, wird nicht mal ausgepackt. Daher wird es wohl nichts (auf die Schnelle) mit einem angepassten Image.

**Reply by fda77** [reply]
Ich bin aber nicht sicher ob dieser Patch so optimal ist, evlt gibt es die Verzeichnisse zu diesem brandring nicht, weshlab wohl
 > This image can be installed on a branded box, but several features might not work (DVB-C, SIP, ..?).

**Reply by AkteX** [reply]
Die genannten Einschränkungen beziehen sich auf das Branding, d.h. dass diese Funktionen entsprechend dem Branding bei der Anwendung des Patches deaktiviert werden.

**Reply by fda77** [reply]
Bis jetzt sind doch nur AVM images bekannt, keine KDG (evtl nur als cvc). Daher wird es wohl das avm branding sein um das es hier geht. Und in dem ist doch eigentlich alles drin

**Comment by fda77** [comment]
Den bootloader muss man wirklich patchen? Sicher? (Geht das überhaupt? Signaturen und so)
Wenn man eh Freetz nutzt könnte man doch Anpassungen vornehmen damit eine freetz-fw auch mit provider-bootloader funktioniert.
Hat schon mal jemand einen avm-bootloader und provider-bootloader ausgelesen und verglichen?
Nix gegen das DEB, aber ich bin recht sicher dass man sowas schonmal im ippf hätte lesen können..
Und was ist "funktioniert nicht" überhaupt?
Und welche Geräte betrifft es? Alle Puma7?

**Comment by AkteX** [comment]
https://www.digital-eliteboard.com/threads/6591-debranding-der-kdg-boxen-anleitung.506639

AVM muss an zwei Stellen reingepatched werden. Der String, welcher durch AVM ersetzt werden muss entspricht dem Branding der Box, ist also nur bei bestimmten Provider-Boxen KDG.

Ohne diesem Patch würde eine originale AVM Firmware ggf. nicht laufen, da das entsprchende Branding nicht enthalten ist.

Weitere Informationen: https://bitbucket.org/fesc2000/ffritz/src/6591/README-6591.md

**Comment by fda77** [comment]
Die Änderungen werden an /dev/mtdblock3 vorgenommen, ich weiss nicht ob das wirklich der bootloader (code) ist, oder nur die normalen "environment" daten (data). 299kb wäre für die config schon viel.
Und das ganze geht nicht über den adam/eva ftp wie bei allen anderen boxen?

Wie dem auch sei, in den meisten avm-fw sind "viel" brandings. Mit Freetz werden ALLE ausser AVM gelöscht (sparte flash). Dies wird dann nach ALL umbenannt. Danach werden Links (ex-Verzeichnisse) von JEDEM Branding auf das ALL gesetzt. 
Ich vermute! mal dass die retail-fw diese KDG branding/oem nicht hat. Wenn ich so eine Box hätte würde ich einen Link von KDG auf das ALL setzten. Und evtl an noch ein paar weiteren stellen. So sollte dann ein Freetz image mit jedem branding im env starten

**Comment by fda77** [comment]
Funktioniert dies mtdblock3 editieren auch für zb 7520 die 1und1 branding hat und sich nicht ändern lässt?

**Reply by AkteX** [reply]
Das weiß ich nicht. Mit 1und1 Branding gibt es nach meiner Kenntnis keine Einschränkungen, welche das Entbranden erforderlich machen würden.

**Reply by fda77** [reply]
Wie gesagt, mit Freetz ist in JEDEM branding der Inhalt von "AVM". Daher gibt es keine unterschiede. Du müsstest schon eine firmware ohne freetz ausprobieren.
Man hat dann nur "1und1" als branding was ein bisschen hässlich ist ^^
